server/src/main/java/org/elasticsearch/reservedstate/service/ReservedStateErrorTask.java

```diff
@@ -50,7 +51,7 @@ ActionListener<ActionResponse.Empty> listener() {
     static boolean isNewError(ReservedStateMetadata existingMetadata, Long newStateVersion) {
         return (existingMetadata == null
             || existingMetadata.errorMetadata() == null
-            || newStateVersion <= 0 // version will be -1 when we can't even parse the file, it might be 0 on snapshot restore
+            || newStateVersion <= RESTORED_VERSION // Long.MIN_VALUE (no version) when we can't parse the file, 0 on snapshot restore
```

Comment:
I wonder, if there are no other cases, if 
`newStateVersion == RESTORED_VERSION || newStateVersion == EMPTY_VERSION` is more clear (you could get rid of the comment, which is usually a good indication)

Reply:
I thought about this as well, and agree it would add clarity. At the same time I'm worried this would easily break things if more magic version ids are introduced. I'll think a bit more about it....

Reply:
Can we enforce the version for error metadata is > 0 when explicitly initialized (ie with a real error)?

Reply:
I had another look at this, I think it's tricky to make any changes on error metadata version as it's coupled to the reserved state version. If a previous reserved state version exists, that one will be the version of the error. However, that can be any of the special versions <= 0.

Reply:
@ldematte changed as suggested